frontend/src/lib/newQueryBuilder/transformQueryBuilderDataModel.ts

```diff
@@ -12,17 +13,47 @@ import { QueryBuilderData } from 'types/common/queryBuilder';
 
 export const transformQueryBuilderDataModel = (
 	data: BuilderQueryDataResourse,
+	query?: QueryBuilderData,
 ): QueryBuilderData => {
 	const queryData: QueryBuilderData['queryData'] = [];
 	const queryFormulas: QueryBuilderData['queryFormulas'] = [];
 
 	Object.entries(data).forEach(([, value]) => {
 		if (FORMULA_REGEXP.test(value.queryName)) {
 			const formula = value as IBuilderFormula;
-			queryFormulas.push({ ...initialFormulaBuilderFormValues, ...formula });
+			const baseFormula = query?.queryFormulas?.find(
+				(f) => f.queryName === value.queryName,
+			);
+			if (!isUndefined(baseFormula)) {
+				// this is part of the flow where we create alerts from dashboard.
+				// we pass the widget query as the base query and accept the expression from the format API response.
+				// which fills the variable values inside the same and is used to create the alerts.
+				// do not accept the full object as the stepInterval field is subject to changes
+				queryFormulas.push({ ...baseFormula, expression: formula.expression });
+			} else {
+				queryFormulas.push({ ...initialFormulaBuilderFormValues, ...formula });
+			}
 		} else {
-			const query = value as IBuilderQuery;
-			queryData.push({ ...initialQueryBuilderFormValuesMap.metrics, ...query });
+			const queryFromData = value as IBuilderQuery;
+			const baseQuery = query?.queryData?.find(
+				(q) => q.queryName === queryFromData.queryName,
+			);
+
+			if (!isUndefined(baseQuery)) {
```

Comment:
Please help me understand this with examples.

Reply:
here what we are doing is taking the base query of the widget as supplied in the dashboard and using the `filters` value from the format API call which fills the values for the variables and only updating the same in the base query.